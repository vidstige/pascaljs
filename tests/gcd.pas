program gcd;
function gcd(a: Integer; b: Integer): Integer;
begin
    while a <> b do
    begin
        if a > b then
           a := a - b
        else
           b := b - a;
    end;
    gcd := a;
end;

begin
    WriteLn('gcd(77, 4) = ', gcd(77, 4));
end.