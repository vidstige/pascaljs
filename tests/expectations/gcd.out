gcd(77, 4) = 1
